plugins/modules/mso_schema_site_anp_epg_bulk_staticport.py

```diff
@@ -0,0 +1,478 @@
+                payload = new_epg
+            else:
+                # If anp in payload, anp exists at site level. Update payload with EPG payload
+                payload["epgs"] = [new_epg]
+    else:
+        # If anp and epg exists at site level
+        epg_path = "{0}/{1}/epgs".format(anp_path, anp)
+
+    ports_path = "{0}/{1}/staticPorts".format(epg_path, epg)
+    staticport_list = []
+
+    if static_ports is not None:
+        for static_port in static_ports:
+            path_type = static_port.get("type") or module_path_type
+            pod = static_port.get("pod") or module_pod
+            leaf = static_port.get("leaf") or module_leaf
+            fex = static_port.get("fex") or module_fex
+            path = static_port.get("path") or module_path  # Note :path has to be diffent for every static port in the list
+            vlan = static_port.get("vlan") or module_vlan
+            primary_micro_segment_vlan = static_port.get("primary_micro_segment_vlan") or module_primary_micro_segment_vlan
+            deployment_immediacy = static_port.get("deployment_immediacy") or module_deployment_immediacy
+            mode = static_port.get("mode") or module_mode
+
+            if path_type == "port" and fex is not None:
+                # Select port path for fex if fex param is used
+                portpath = "topology/{0}/paths-{1}/extpaths-{2}/pathep-[{3}]".format(pod, leaf, fex, path)
+            elif path_type == "vpc":
+                portpath = "topology/{0}/protpaths-{1}/pathep-[{2}]".format(pod, leaf, path)
+            else:
+                portpath = "topology/{0}/paths-{1}/pathep-[{2}]".format(pod, leaf, path)
+
+            # Get Leaf
+            # If anp at site level and epg is at site level
+            if "anpRef" not in payload and "epgRef" not in payload:
+                portpaths = [p.get("path") for p in mso_objects.get("site_anp_epg").details.get("staticPorts")]
+                if portpath in portpaths:
+                    portpath_idx = portpaths.index(portpath)
+                    mso.existing = mso_objects.get("site_anp_epg").details.get("staticPorts")[portpath_idx]
+
+            if state == "query":
+                if leaf is None or vlan is None:
+                    mso.existing = mso_objects.get("site_anp_epg").details.get("staticPorts")
+                elif not mso.existing:
+                    mso.fail_json(msg="Static port '{portpath}' not found".format(portpath=portpath))
+                mso.exit_json()
+
+            new_leaf = dict(
+                deploymentImmediacy=deployment_immediacy,
+                mode=mode,
+                path=portpath,
+                portEncapVlan=vlan,
+                type=path_type,
+            )
+
+            if primary_micro_segment_vlan:
+                new_leaf.update(microSegVlan=primary_micro_segment_vlan)
+
+            # validate and append staticports to staticport_list if path variable is different
+            unique_paths = []
```

Comment:
should this be defined before the for loop? else it would reset always before the if statement